### DISCUSSION TITLE
These updates encountered an error and will be retried. Click a checkbox below to force a retry now.

### DISCUSSION THREAD

**Discussion by andorfermichael** [renovatebot/renovate#10680] [Request Help] [answered]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [x] GitLab self-hosted

Renovate version: 25.49

**Describe the bug**

Dashboard says: `These updates encountered an error and will be retried. Click a checkbox below to force a retry now.`

This issue contains a list of Renovate updates and their statuses.

## Awaiting Schedule

These updates are awaiting their schedule. Click on a checkbox to get an update now.
 - [ ] chore(deps): update dependency @nuxtjs/tailwindcss to v3.4.3
 - [ ] chore(deps): update dependency husky to v4.3.8
 - [ ] chore(deps): update dependency lint-staged to v10.5.4
 - [ ] fix(deps): update dependency @nuxtjs/pwa to v3.3.5
 - [ ] chore(deps): update dependency @babel/runtime-corejs3 to v7.14.7
 - [ ] chore(deps): update dependency eslint to v7.29.0
 - [ ] chore(deps): update dependency eslint-config-prettier to v7.2.0
 - [ ] chore(deps): update dependency eslint-plugin-prettier to v3.4.0
 - [ ] chore(deps): update dependency fs-extra to v9.1.0
 - [ ] chore(deps): update dependency prettier to v2.3.2
 - [ ] chore(deps): update dependency sass to v1.35.1
 - [ ] chore(deps): update dependency sass-loader to v10.2.0
 - [ ] chore(deps): update dependency stylelint to v13.13.1
 - [ ] chore(deps): update dependency stylelint-scss to v3.19.0
 - [ ] chore(deps): update nuxtjs monorepo to v2.15.7 (`@nuxt/types`, `nuxt`)
 - [ ] chore(deps): update vue monorepo (`@vue/test-utils`, `vue-jest`)
 - [ ] fix(deps): update dependency @nuxt/content-theme-docs to v0.10.2
 - [ ] fix(deps): update dependency @nuxtjs/axios to v5.13.6
 - [ ] fix(deps): update dependency core-js to v3.15.2
 - [ ] fix(deps): update dependency d3 to v6.7.0
 - [ ] fix(deps): update dependency date-fns to v2.22.1
 - [ ] fix(deps): update dependency vue2-leaflet to v2.7.1
 - [ ] chore(deps): update commitlint monorepo to v12 (major) (`@commitlint/cli`, `@commitlint/config-conventional`)
 - [ ] chore(deps): update dependency @nuxtjs/eslint-config to v6
 - [ ] chore(deps): update dependency eslint-config-prettier to v8
 - [ ] chore(deps): update dependency fs-extra to v10
 - [ ] chore(deps): update dependency lint-staged to v11
 - [ ] chore(deps): update dependency sass-loader to v12
 - [ ] chore(deps): update dependency stylelint-config-standard to v22
 - [ ] chore(deps): update jest monorepo to v27 (major) (`babel-jest`, `jest`)
 - [ ] fix(deps): update dependency @nuxtjs/sentry to v5
 - [ ] fix(deps): update dependency d3 to v7
 - [ ] fix(deps): update dependency vue-matomo to v4

## Errored

These updates encountered an error and will be retried. Click a checkbox below to force a retry now.

 - [ ] chore(deps): pin dependencies (`@babel/runtime-corejs3`, `@commitlint/cli`, `@commitlint/config-conventional`, `@nuxt/content-theme-docs`, `@nuxt/types`, `@nuxtjs/axios`, `@nuxtjs/eslint-config`, `@nuxtjs/eslint-module`, `@nuxtjs/pwa`, `@nuxtjs/sentry`, `@nuxtjs/stylelint-module`, `@nuxtjs/tailwindcss`, `@vue/test-utils`, `babel-eslint`, `babel-jest`, `cloudflare`, `core-js`, `d3`, `date-fns`, `eslint`, `eslint-config-prettier`, `eslint-plugin-nuxt`, `eslint-plugin-prettier`, `fibers`, `fs-extra`, `hashicorp/terraform`, `hcloud`, `husky`, `jest`, `leaflet`, `lint-staged`, `nuxt`, `postcss-focus-within`, `prettier`, `sass`, `sass-loader`, `stylelint`, `stylelint-config-prettier`, `stylelint-config-standard`, `stylelint-order`, `stylelint-scss`, `topojson`, `vue-jest`, `vue-matomo`, `vue2-leaflet`, `workerize-loader`)

However, retry does not work.

MR creation does not work for the project for `company/map project`, it works however for other projects such as `company/wpc `or `company/nuxt`.

Runner logs do not print any error.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
Running with gitlab-runner 14.0.1 (c1edb478)
  on infra1 renovate maeesMZL
Resolving secrets
00:00
Preparing the "docker" executor
Using Docker executor with image renovate/renovate:25.49@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b ...
Starting service docker:20.10.7-dind ...
Pulling docker image docker:20.10.7-dind@sha256:4e1e22f471afc7ed5e024127396f56db392c1b6fc81fc0c05c0e072fb51909fe ...
Using docker image sha256:b0a9f712488ae3b3ddf576e14e56186f709e4735fe042b01f76521878929d535 for docker:20.10.7-dind@sha256:4e1e22f471afc7ed5e024127396f56db392c1b6fc81fc0c05c0e072fb51909fe with digest docker@sha256:4e1e22f471afc7ed5e024127396f56db392c1b6fc81fc0c05c0e072fb51909fe ...
Waiting for services to be up and running...
Pulling docker image renovate/renovate:25.49@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b ...
Using docker image sha256:bdaf1372959bb42cc79628cf092ee3600e15387562120c140cb7508807e84e58 for renovate/renovate:25.49@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b with digest renovate/renovate@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b ...
Preparing environment
00:01
Running on runner-maeesmzl-project-199-concurrent-0 via infra1.company.com...
Getting source from Git repository
Fetching changes with git depth set to 50...
Reinitialized existing Git repository in /builds/company/renovate-runner/.git/
Checking out af70d19d as main...
Removing node_modules/
Removing renovate-log.ndjson
Removing renovate/
Skipping Git submodules setup
Restoring cache
Checking cache for main-renovate...
No URL provided, cache will not be downloaded from shared cache server. Instead a local version of cache will be extracted. 
Successfully extracted cache
Executing "step_script" stage of the job script
Using docker image sha256:bdaf1372959bb42cc79628cf092ee3600e15387562120c140cb7508807e84e58 for renovate/renovate:25.49@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b with digest renovate/renovate@sha256:56def454410e3a44d20081244401330e58c1ce417dff3237f34dd3a20d4a218b ...
$ renovate $RENOVATE_EXTRA_FLAGS
 INFO: Repository started (repository=company/company-nuxt)
       "renovateVersion": "25.49.0"
 INFO: Dependency extraction complete (repository=company/company-nuxt)
       "baseBranch": "master",
       "stats": {
         "managers": {"npm": {"fileCount": 1, "depCount": 44}},
         "total": {"fileCount": 1, "depCount": 44}
       }
 INFO: Repository finished (repository=company/company-nuxt)
       "durationMs": 14115
 INFO: Repository started (repository=company/wpc)
       "renovateVersion": "25.49.0"
 INFO: Dependency extraction complete (repository=company/wpc)
       "baseBranch": "master",
       "stats": {
         "managers": {"npm": {"fileCount": 1, "depCount": 24}},
         "total": {"fileCount": 1, "depCount": 24}
       }
 INFO: Repository finished (repository=company/wpc)
       "durationMs": 3869
 INFO: Repository started (repository=company/map)
       "renovateVersion": "25.49.0"
 INFO: Dependency extraction complete (repository=company/map)
       "baseBranch": "master",
       "stats": {
         "managers": {
           "npm": {"fileCount": 2, "depCount": 45},
           "terraform": {"fileCount": 5, "depCount": 13}
         },
         "total": {"fileCount": 7, "depCount": 58}
       }
 INFO: Repository finished (repository=company/map)
       "durationMs": 50599
Saving cache for successful job
Creating cache main-renovate...
/builds/company/renovate-runner/renovate: found 68462 matching files and directories 
No URL provided, cache will be not uploaded to shared cache server. Cache will be stored only locally. 
Created cache
Uploading artifacts for successful job
00:01
Uploading artifacts...
renovate-log.ndjson: found 1 matching files and directories 
Uploading artifacts as "archive" to coordinator... ok  id=89639 responseStatus=201 Created token=xY6K-gCg
Cleaning up file based variables
00:01
Job succeeded
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

My configs:

```
// renovate-config-base
{
  "extends": [
    "config:base",
    ":timezone(Europe/Vienna)",
    ":label(Renovate)",
    ":assignee(gitlab)"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["major"],
      "addLabels": ["SemVer::major"]
    },
    {
      "matchUpdateTypes": ["minor"],
      "addLabels": ["SemVer::minor"]
    },
    {
      "matchUpdateTypes": ["patch"],
      "addLabels": ["SemVer::patch"]
    }
  ]
}
```

```
// renovate-config-javascript
{
  "extends": [
    "local>company/renovate-config-base",
    ":pinAllExceptPeerDependencies",
    ":onlyNpm"
  ],
  "packageRules": [
    {
      "matchDatasources": ["npm"],
      "stabilityDays": 3
    },
    {
      "matchDepTypes": ["dependencies"],
      "addLabels": ["Dependency::runtime"]
    },
    {
      "matchDepTypes": ["devDependencies"],
      "addLabels": ["Dependency::development"]
    },
    {
      "matchDepTypes": ["peerDependencies"],
      "addLabels": ["Dependency::peer"]
    },
    {
      "matchDepTypes": ["optionalDependencies"],
      "addLabels": ["Dependency::optional"]
    },
    {
      "matchDepTypes": ["bundledDependencies"],
      "addLabels": ["Dependency::bundled"]
    }
  ]
}
```
```
// company/map/renovate.json
{
    "extends": [
        "local>company/renovate-config-javascript",
        ":disableRateLimiting",
        "schedule:weekly",
        ":dependencyDashboard"
    ],
    "packageRules": [
        {
            "matchPackageNames": ["@nuxtjs/tailwindcss"],
            "allowedVersions": "<4"
        },
        {
            "matchPackageNames": ["husky"],
            "allowedVersions": "<5"
        },
        {
            "matchPackageNames": ["core-js"],
            "allowedVersions": "<4"
        },
        {
            "matchPackageNames": ["postcss-focus-within"],
            "allowedVersions": "<5"
        }
    ],
    "dependencyDashboardAutoclose": true
}
```

```
// company/map/package.json
{
    "name": "map",
    "version": "1.0.0",
    "private": true,
    "scripts": {
        "dev": "nuxt",
        "build": "nuxt build",
        "start": "nuxt start",
        "generate": "nuxt generate",
        "lint:js": "eslint --ext .js,.vue --ignore-path .gitignore .",
        "lint:style": "stylelint **/*.{vue,css} --ignore-path .gitignore",
        "lint": "yarn lint:js && yarn lint:style",
        "lint:fix": "yarn lint:js --fix && yarn lint:style --fix",
        "test": "jest"
    },
    "lint-staged": {
        "*.{js,vue}": "eslint",
        "*.{css,vue}": "stylelint"
    },
    "husky": {
        "hooks": {
            "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
            "pre-commit": "lint-staged"
        }
    },
    "dependencies": {
        "@nuxtjs/axios": "^5.12.4",
        "@nuxtjs/pwa": "^3.3.3",
        "@nuxtjs/sentry": "^4.5.0",
        "core-js": "3",
        "d3": "^6.3.1",
        "date-fns": "^2.16.1",
        "leaflet": "^1.7.1",
        "nuxt": "^2.14.12",
        "topojson": "^3.0.2",
        "vue-matomo": "^3.14.0-0",
        "vue2-leaflet": "^2.6.0"
    },
    "devDependencies": {
        "@babel/runtime-corejs3": "^7.12.5",
        "@commitlint/cli": "^11.0.0",
        "@commitlint/config-conventional": "^11.0.0",
        "@nuxt/types": "~2.14.0",
        "@nuxtjs/eslint-config": "^5.0.0",
        "@nuxtjs/eslint-module": "^3.0.2",
        "@nuxtjs/stylelint-module": "^4.0.0",
        "@nuxtjs/tailwindcss": "^3.4.1",
        "@vue/test-utils": "^1.1.2",
        "babel-core": "7.0.0-bridge.0",
        "babel-eslint": "^10.1.0",
        "babel-jest": "^26.6.3",
        "eslint": "^7.16.0",
        "eslint-config-prettier": "^7.1.0",
        "eslint-plugin-nuxt": "^2.0.0",
        "eslint-plugin-prettier": "^3.3.0",
        "fibers": "^5.0.0",
        "fs-extra": "^9.0.1",
        "husky": "^4.3.6",
        "jest": "^26.6.3",
        "lint-staged": "^10.5.3",
        "prettier": "^2.2.1",
        "sass": "^1.30.0",
        "sass-loader": "^10.1.0",
        "stylelint": "^13.8.0",
        "stylelint-config-prettier": "^8.0.2",
        "stylelint-config-standard": "^20.0.0",
        "stylelint-order": "^4.1.0",
        "stylelint-scss": "^3.18.0",
        "vue-jest": "^4.0.0-rc.1",
        "workerize-loader": "^1.3.0"
    },
    "browserslist": [
        "last 2 versions",
        "> 0.15%",
        "not IE 9",
        "chrome >70",
        "android >4",
        "ios_saf >9"
    ],
    "resolutions": {
        "postcss-focus-within": "^4.0.0"
    }
}
```

...

**Comment by rarkins** [comment, marked as answer]
Needs debug logs and should be a discussion instead of an issue.

**Reply by rarkins** [reply]
Run with LOG_LEVEL=debug and capture a run where the MR is retried after you tick the checkbox

**Reply by andorfermichael** [reply]
Thanks. Sure, actually I thought I had already activated the log, but I accidentally reversed variable name with value :see_no_evil:

**Reply by andorfermichael** [reply]
Here is the log:

[renovate-log.txt](https://github.com/renovatebot/renovate/files/6746001/renovate-log.txt)

**Reply by rarkins** [reply]
Please look through the log and extract just the relevant parts (eg the branch name in question)

**Reply by viceice** [reply]
```
DEBUG: Converting git error to ExternalHostError (repository=company/map, branch=renovate/pin-dependencies)
       "err": {
         "task": {
           "commands": [
             "push",
             "origin",
             "renovate/pin-dependencies:renovate/pin-dependencies",
             "--force",
             "-u",
             "--no-verify",
             "--verbose",
             "--porcelain"
           ],
           "format": "utf-8"
         },
         "message": "To https://**redacted**@renovateapp.com' is not a member of team\nerror: failed to push some refs to 'https://git.company.com/company/map.git'\n",
         "stack": "Error: To https://**redacted**@renovateapp.com' is not a member of team\nerror: failed to push some refs to 'https://git.company.com/company/map.git'\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
       }
```
There is a clear message what happens. Next time trim the logs to at least the failing repo and not the full log.

It seems the renovate user has not enough rights on the `company/map` repo

**Reply by andorfermichael** [reply]
Sorry I will trim next time. I saw the message but I didn't find it quite helpful.

Nevertheless the solution was to go to Settings/Repository/Push Rules and uncheck:

**Check whether the commit author is a GitLab user**
- [ ]  Restrict commits to existing GitLab users.

**Reply by rarkins** [reply]
@viceice I think we could try to catch this and raise a config error

**Reply by viceice** [reply]
Sounds good, can send a PR later

**Reply by viceice** [reply]
#10731
